src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp

```diff
@@ -458,10 +504,41 @@ bool YarpRobotControl::initialize(std::weak_ptr<ParametersHandler::IParametersHa
         return false;
     }
 
-    bool ok = ptr->getParameter("positioning_duration", m_pimpl->positioningDuration);
-    ok = ok && ptr->getParameter("positioning_tolerance", m_pimpl->positioningTolerance);
+    // optional parameters
+    int temp = 0;
+    if (ptr->getParameter("reading_timeout", temp))
```

Comment:
The problem with this is that if the ``reading_timeout`` field is missing, it will always throw an error. We may need to add a "quiet" mode for optional parameters. ANyways, this goes outside the scope of this PR.

Reply:
> The problem with this is that if the reading_timeout field is missing, it will always throw an error. We may need to add a "quiet" mode for optional parameters. Anyways, this goes outside the scope of this PR.

Yes, this is really annoying. I opened an issue for that: https://github.com/dic-iit/bipedal-locomotion-framework/issues/120